lib/matplotlib/scale.py

```diff
@@ -562,40 +557,31 @@ class SymmetricalLogScale(ScaleBase):
     InvertedSymmetricalLogTransform = InvertedSymmetricalLogTransform
 
     def __init__(self, axis, **kwargs):
-        if axis.axis_name == 'x':
-            base = kwargs.pop('basex', 10.0)
-            linthresh = kwargs.pop('linthreshx', 2.0)
-            subs = kwargs.pop('subsx', None)
-            linscale = kwargs.pop('linscalex', 1.0)
-        else:
-            base = kwargs.pop('basey', 10.0)
-            linthresh = kwargs.pop('linthreshy', 2.0)
-            subs = kwargs.pop('subsy', None)
-            linscale = kwargs.pop('linscaley', 1.0)
-        if kwargs:
-            warn_deprecated(
-                '3.2.0',
-                message=(
-                    f"SymmetricalLogScale got an unexpected keyword "
-                    f"argument {next(iter(kwargs))!r}. "
-                    'In the future this will raise TypeError')
-            )
-            # raise TypeError(f"SymmetricalLogScale got an unexpected keyword "
-            #                 f"argument {next(iter(kwargs))!r}")
-
-        if base <= 1.0:
-            raise ValueError("'basex/basey' must be larger than 1")
-        if linthresh <= 0.0:
-            raise ValueError("'linthreshx/linthreshy' must be positive")
-        if linscale <= 0.0:
-            raise ValueError("'linscalex/linthreshy' must be positive")
-
+        rename_parameter = cbook._rename_parameter
+
+        @rename_parameter("3.2", f"base{axis.axis_name}", "base")
+        @rename_parameter("3.2", f"linthresh{axis.axis_name}", "linthresh")
+        @rename_parameter("3.2", f"subs{axis.axis_name}", "subs")
+        @rename_parameter("3.2", f"linscale{axis.axis_name}", "linscale")
+        def __init__(*, base=10, linthresh=2, subs=None, linscale=1, **kwargs):
```

Comment:
any particular reason to call this `__init__`? I would just use `init` because it's just a regular local method. The dunder may be a bit misleading.

Reply:
Yes, this is so that when the wrong arg names are passed, Python generates the error message with the right name (`__init__() got an unexpected argument...`).

Reply:
Maybe a comment in the code for why you did this?  I guess its just a  place holder until a future version?